go/exocom-mock/exocom-mock_test.go

```diff
@@ -0,0 +1,156 @@
+package exocomMock_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/DATA-DOG/godog/gherkin"
+	"github.com/Originate/exocom/go/exocom-mock"
+	"github.com/Originate/exocom/go/structs"
+	"github.com/Originate/exocom/go/utils"
+	"github.com/gorilla/websocket"
+	"github.com/phayes/freeport"
+)
+
+func newExocom(port int) *exocomMock.ExoComMock {
+	exocom := exocomMock.New()
+	go func() {
+		err := exocom.Listen(port)
+		if err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+	return exocom
+}
+
+// nolint gocyclo
+func FeatureContext(s *godog.Suite) {
+	var exocomPort int
+	var exocom *exocomMock.ExoComMock
+	var socket *websocket.Conn
+
+	s.BeforeScenario(func(interface{}) {
+		exocomPort = freeport.GetPort()
```

Comment:
This code is okay. FYI Go provides the https://golang.org/pkg/net/http/httptest package to make testing HTTP-based server easier. Maybe we can use it somewhere.